web/src/server_events_dispatch.js

```diff
@@ -667,7 +667,6 @@ export function dispatch_normal_event(event) {
             }
             if (event.property === "demote_inactive_streams") {
                 stream_list.update_streams_sidebar();
-                stream_data.set_filter_out_inactives();
```

Comment:
This likely needs to be changed.

Reply:
I am not sure what needs to be changed here, I think we might not need to call this function here because it doesn't update anything other than a variable which I removed in the new code.

Reply:
OK, I see what happened here. So, you decided to remove the `filter_out_inactives` variable, and that removed the need for this line.

In general, you should always make that sort of independent but adjacent refactoring change in its own commit from a commit that just moves code -- we discuss this explicitly in our commit guidelines, and I think this PR in particular is a great example of why that is an important development habit. First, it's way harder to read a change with this sort of refactoring squashed into a change that's just moving code -- moving code produces giant diffs because everything moved, and that obscures any actual changes. (Read about the `--color-moved` option to `git diff` to see why).

Second, that change is actually a bug, and having it separate would have helped a reviewer focus on it and notice it.  In particular, the reason that variable existed was as a cache of the calculation of whether there's 30+ streams; without it, rendering the stream list has quadratic behavior such that being subscribed to 1000 streams will lock up your computer with the "automatic" setting. You could probably have found this with `git log -Sfilter_out_inactives` and reading the commit that introduced it.

In any case, please undo that change, and let's add a comment explaining why we have this weird cache of that calculation.

Reply:
Undid the change and left a comment, I fell for that refactor because it looked very obvious to me to remove that variable at first glance but I really should have done that refactoring in a separate commit because it forced me to do a fairly time taking rebase, I guess I learned this the hard way haha, I will keep in mind to not let myself do refactorings in the same commit as the migration one! Thanks again for the review!